### PR TITLE
Fix SelectList and RichText reactivity problems

### PR DESCRIPTION
**Pull request by caleeli**
## Issue & Reproduction Steps
There are reactivity problems with multiple components like:
- SelectList
- RichText 

## Solution
Most of the reactivity problems are caused by watching `data`, `validationData`, `_parent` properties and using the option `deep: true`. The code included in this PR modify this problems in components:
- SelectList
- RichText 

## How to Test
- Test with large screens, with or without dependent fields
- Go to preview
- Change some value of any field
- This triggers reactivity of many if not all the elements of the screen

## Related Tickets & Packages
- Required by: https://github.com/ProcessMaker/screen-builder/pull/1258
- https://processmaker.atlassian.net/browse/FOUR-6760

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

### CODE REVIEW

File: src/components/FormHtmlEditor.vue [hunk 1]
```diff
@@ -1,74 +1,92 @@
...
         v-if="!$attrs.disabled"
-        v-on="$listeners"
-        v-bind="$attrs"
         :value="rendered"
         :init="editorSettings"
+        v-bind="$attrs"
+        v-on="$listeners"
       />
-      <div v-else v-html="rendered"></div>
+      <div v-else>{{ rendered }}</div>
```
Comment (josechirivella) [inline src/components/FormHtmlEditor.vue:11]: was this change intended?

File: src/components/FormSelectList.vue [hunk 2]
```diff
@@ -50,390 +51,414 @@
...
+    // Special Case String Value:
+    //   Review test tests/e2e/specs/MultiselectWithStringValue.spec.js in ScreenBuilder
+    const resetValueIfNotInOptions = typeof this.value !== "string";
+    this.fillSelectListOptions(resetValueIfNotInOptions);
+  },
+  methods: {
+    async loadOptionsFromDataConnector(options) {
+      const { selectedEndPoint } = options;
+      const { selectedDataSource } = options;
+      const { dataName } = options;
```
Comment (josechirivella) [inline src/components/FormSelectList.vue:189]:
```suggestion
      const { selectedEndPoint, selectedDataSource, dataName } = options;
```

File: src/components/FormSelectList.vue [hunk 3]
```diff
@@ -50,390 +51,414 @@
...
-            const pmql = Mustache.render(this.options.pmqlQuery, {data: this.validationData});
-            params.config.outboundConfig = [
-              { type: 'PARAM', key: 'pmql', value: pmql }
-            ];
-          }
+      // If no data source has been specified, do not make the api call
+      if (
+        selectedDataSource === null ||
+        typeof selectedDataSource === "undefined" ||
+        selectedDataSource.toString().trim().length === 0
```
Comment (josechirivella) [inline src/components/FormSelectList.vue:195]: this could be refactored to
```suggestion
!!selectedDataSource && selectedDataSource.toString().trim().length === 0
```

File: src/components/FormSelectList.vue [hunk 4]
```diff
@@ -50,390 +51,414 @@
...
-              })
-              .catch(err => {
-                /* Ignore error */
-              });
-        }, 700)
+      // If no endpoint has been specified, do not make the api call
+      if (
+        selectedEndPoint === null ||
+        typeof selectedEndPoint === "undefined" ||
+        selectedEndPoint.toString().trim().length === 0
```
Comment (josechirivella) [inline src/components/FormSelectList.vue:209]: this could be refactored to
```suggestion
!!selectedDataSource && selectedDataSource.toString().trim().length === 0
```

File: src/components/FormSelectList.vue [hunk 5]
```diff
@@ -50,390 +51,414 @@
...
+          params
+        );
+        const list = dataName ? get(response.data, dataName) : response.data;
+        const transformedList = this.transformOptions(list);
+        this.$root.$emit("selectListOptionsUpdated", transformedList);
+        this.selectListOptions = transformedList;
+        return true;
+      } catch (err) {
+        /* Ignore error */
+        console.warn(err);
```
Comment (josechirivella) [inline src/components/FormSelectList.vue:250]:
```suggestion
        console.error(err);
```

**Review by josechirivella — CHANGES_REQUESTED** [review]

File: src/components/FormHtmlEditor.vue (hunk 1 above)
Comment (caleeli) [reply src/components/FormHtmlEditor.vue:11]: No, that's a mistake

Comment (rodriquelca) [reply src/components/FormHtmlEditor.vue]: this change was reverted

File: src/components/FormSelectList.vue (hunk 2 above)
Comment (rodriquelca) [reply src/components/FormSelectList.vue]: done

File: src/components/FormSelectList.vue (hunk 3 above)
Comment (rodriquelca) [reply src/components/FormSelectList.vue]: done

File: src/components/FormSelectList.vue (hunk 4 above)
Comment (rodriquelca) [reply src/components/FormSelectList.vue]: done but with selectedEndPoint variable

File: src/components/FormSelectList.vue (hunk 5 above)
Comment (rodriquelca) [reply src/components/FormSelectList.vue]: done

**Review by josechirivella — APPROVED** [review]
